app/models/work_package/pdf_export/common/attachments.rb

```diff
@@ -70,8 +70,21 @@ def attachment_image_filepath(work_package, src)
     resize_image(local_file.path)
   end
 
-  def attachment_by_api_content_src(work_package, src)
-    # find attachment by api-path
-    work_package.attachments.find { |a| api_url_helpers.attachment_content(a.id) == src }
+  def attachment_by_api_content_src(_work_package, src)
+    # /attachments/#{id}/content
+    parts = src.split("/")
+    attachments_id = parts[-2]
+    # return nil if the src is not an attachment url
+    return nil unless api_url_helpers.attachment_content(attachments_id) == src
```

Comment:
WDYT about instead of splitting here, we use a regex:

```ruby
attachment_regex = %r{/attachments/(\d+)/content}

return nil if src !~ attachment_regex
attachments_id = src.scan(attachment_regex).first.first
```